packages/plugin-legacy/src/index.ts

```diff
@@ -324,31 +324,23 @@ function viteLegacyPlugin(options: Options = {}): Plugin[] {
           | string
           | ((chunkInfo: PreRenderedChunk) => string)
           | undefined,
-        defaultFileName = '[name]-legacy-[hash].js',
+        defaultFileName = '[name]-[hash]-legacy.js',
       ): string | ((chunkInfo: PreRenderedChunk) => string) => {
         if (!fileNames) {
           return path.posix.join(config.build.assetsDir, defaultFileName)
         }
 
         return (chunkInfo) => {
-          let fileName =
+          const fileName =
             typeof fileNames === 'function' ? fileNames(chunkInfo) : fileNames
 
-          if (fileName.includes('[name]')) {
-            // [name]-[hash].[format] -> [name]-legacy-[hash].[format]
-            fileName = fileName.replace('[name]', '[name]-legacy')
-          } else if (fileName.includes('[hash]')) {
-            // custom[hash].[format] -> [name]-legacy[hash].[format]
-            // custom-[hash].[format] -> [name]-legacy-[hash].[format]
-            // custom.[hash].[format] -> [name]-legacy.[hash].[format]
-            fileName = fileName.replace(/[.-]?\[hash\]/, '-legacy$&')
```

Comment:
Yes, we should only handle the case that ends up with `-legacy` as @bluwy explains. I think we should only add an extra case here:
```js
fileName = fileName.startsWith('[hash]') ? `legacy-${fileName}` : fileName.replace(/[.-]?\[hash\]/, '-legacy$&')
```

The comments in this section are wrong, you could also fix that as you did in the current form of the PR.

Reply:
> fileName = fileName.startsWith('[hash]') ? `legacy-${fileName}` : fileName.replace(/[.-]?\[hash\]/, '-legacy$&')

This won't solve the problem，eg: `assets/[hash].js`

Reply:
Ah, you are right. Ok, `xxx/[hash]` should also be changed 👍🏼